frontend/src/views/Join.tsx

```diff
@@ -1,28 +1,92 @@
-import React, { useState } from 'react';
-import { PrimaryButton } from '../components/core/Button';
-import Input from '../components/core/Input';
-import {
-  connect, disconnect, sendUser, SOCKET_ENDPOINT,
-} from '../api/Socket';
+import React, { useState, useEffect, ReactElement } from 'react';
+import { LargeText, Text } from '../components/core/Text';
+import { LargeCenterInputText, LargeInputButton } from '../components/core/Input';
+import { connect, SOCKET_ENDPOINT } from '../api/Socket';
 
 function JoinGamePage() {
   // Declare nickname state variable.
   const [nickname, setNickname] = useState('');
 
-  return (
-    <div>
-      <Input placeholder="Enter your nickname" onChange={(event) => setNickname(event.target.value)} />
-      <PrimaryButton onClick={() => connect(SOCKET_ENDPOINT, nickname)}>
-        Connect
-      </PrimaryButton>
-      <PrimaryButton onClick={disconnect}>
-        Disconnect
-      </PrimaryButton>
-      <PrimaryButton onClick={() => sendUser(nickname)}>
-        Send User Information
-      </PrimaryButton>
-    </div>
-  );
+  /**
+   * The nickname is valid if it is non-empty, has no spaces, and
+   * is <= 16 characters. This is updated whenever the nickname changes.
+   */
+  const [validNickname, setValidNickname] = useState(false);
+  useEffect(() => {
+    setValidNickname(nickname.length > 0 && !nickname.includes(' ') && nickname.length <= 16);
+  }, [nickname]);
+
+  // Variable to hold whether the user is focused on the text input field.
+  const [focusInput, setFocusInput] = useState(false);
+
+  /**
+   * Stores the current page state, where:
+   * 0 = Enter room ID state (currently unused)
+   * 1 = Enter nickname state
+   * 2 = Waiting room state
+   */
+  const [pageState, setPageState] = useState(1);
+
+  // Create variable to hold the "Join Page" content.
+  let joinPageContent: ReactElement | undefined;
+
+  switch (pageState) {
```

Comment:
I'm open to changing the design of this portion of the file if it is too complex, let me know your thoughts (I'm also happy to add more comments)!

Reply:
I think maybe the waiting room component might eventually be moved to another page, but for the sake of this PR and the waiting room/join room page designs, I think this looks great